### ISSUE TITLE
trustedOrigins is not respected

### ISSUE THREAD

**Issue by jakst** [better-auth/better-auth#605]
**Describe the bug**
Adding an origin to trustedOrigins does not have any effect.

```ts
import { betterAuth } from "better-auth"
import Database from "better-sqlite3"

export const auth = betterAuth({
  trustedOrigins: ["http://localhost:5173"],
  database: new Database("./sqlite.db"),
})
```

Requests to sign in from the frontend running at `http://localhost:5173` are denied with the following error message.

```
 ERROR   Invalid origin: http://localhost:5173

ℹ  If it's a valid URL, please add http://localhost:5173 to trustedOrigins in your auth config
 Current list of trustedOrigins: http://localhost:3000
 ```
 
 The interesting thing is that the error message adds an extra instance of `http://localhost:3000` to the list for each request. So this is what it looks like when doing three requests after each other.
 
 ```
 ERROR   Invalid origin: http://localhost:5173

ℹ  If it's a valid URL, please add http://localhost:5173 to trustedOrigins in your auth config
 Current list of trustedOrigins: http://localhost:3000

 ERROR   Invalid origin: http://localhost:5173

ℹ  If it's a valid URL, please add http://localhost:5173 to trustedOrigins in your auth config
 Current list of trustedOrigins: http://localhost:3000,http://localhost:3000

 ERROR   Invalid origin: http://localhost:5173

ℹ  If it's a valid URL, please add http://localhost:5173 to trustedOrigins in your auth config
 Current list of trustedOrigins: http://localhost:3000,http://localhost:3000,http://localhost:3000
 ```

**To Reproduce**

1. Set up a better-auth server on port 3000. I was following the tutorial for Hono.
2. Add `trustedOrigins: ["http://localhost:5173"]` to the config
3. Set up a frontend on port 5173 and add a signin button that uses the better-auth SDK.
4. Try to signin
5. See the error in the auth server logs

**Expected behavior**
I should be able to trust different origins than the one that the auth server is running on.

**Desktop (please complete the following information):**

- OS: MacOS v15.1
- Browser Chrome 
- Version better-auth v1.0.0-canary.3

**Comment by teukuamru** [comment]
i can confirm this also happens in version 0.8.8

**Comment by jakst** [comment]
This was fixed in `v1.0.0-canary.6`.
